### PR TITLE
Update readme

### LINKED ISSUE #187: Mention about Golang  client in project README
Hey everybody,

This isn't really an issue but I don't know where to share this. I just published a Go SDK for the eWallet and Admin API: https://github.com/Alainy/OmiseGo-Go-SDK

It isn't a finished package yet but I hope it can be of use for someone. Also feedback is welcome!

### LINKED ISSUE #253: Improve project README
The current README for the project is a little big overwhelming to get started. I believe the user will benefit greatly if we simplify the README to just topic user will first look for, and link to a more advanced topics in the docs/ directory. 

Here's some example of a project I stumbled upon that has a great README or something we could take hint from:

* [FoundationDB](https://github.com/apple/foundationdb) — very concise and clear, and has everything I want to know about the project at a glance. I especially like the "Compiling from source" which is loosely written, but very easy to follow.
* [Milksnake](https://github.com/getsentry/milksnake) — this is organized in a "Why", "What", and "How" which I like. I think making the heading a question make it easy to focus on explaining a single thing, and it coincides nicely with the user's expectation of the README in the first place, e.g. if we have "How to install?" section for when user look for how to install. 
* [Re-frame](https://github.com/Day8/re-frame) — a bit different from what I'm proposing for README, and it's more of an article, but the framework require full understanding of the concept and the author did a great job putting all pieces together that you'd be able to use it immediately after finished reading the README.

Feel free to add more examples.

### PR DESCRIPTION
**Pull request by unnawut**
Issue/Task Number: T481
Closes #253
Closes #187

# Overview

This PR updates the readme and all other full text documentation.

The easiest way to view the new version is through the branch view at: https://github.com/omisego/ewallet/tree/t481-readme

# Changes

- Updated the main README.md
- Moved a lot of the previous README.md content into separate own documents

# Implementation Details

Here are some principles I set while making this version of the README.

- The content is one-directional. Minimal or no skipping is required.
- Each section in the main readme is half a laptop page in height or less.
- Each section contains no more than 2 paragraphs.
- Any decision branching is clearly stated with call to action (pick one, choose ..., etc.) before the branching options.
- If content contains decision branches, the branches are in short 1-2 line bullet point. No more than 3 branches allowed.
- Any longer content that are decision branches are clearly labeled in the title e.g. `Option 1: ...`, `Option 2: ...` (see Docker Setup page for example)
- When branched off to another page, the reader can complete that page, and come back to continue right where they left off.

# Usage

Read, read, read. The easiest way to view the new version is through the branch view at: https://github.com/omisego/ewallet/tree/t481-readme

# Impact

No DB or API changes.

### CODE REVIEW

**Comment by unnawut** [comment]
Hey everyone. This is still a work in progress but I've put in quite some chunk of edit, and I don't want to proceed without getting some initial feedbacks. Please let me know if you have any opinions about the new main README.md.

The subsequent pages havn't really been edited yet, but at least you should see the correct flow from the README.md to other pages. Let me know if you have any suggestion be it structure or content wise!

**Comment by unnawut** [comment]
Ready for review. I think this is the best I can do for now given the time already spent on it.

File: docs/guides/api_responsibilities.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# Integrating the OmiseGO SDK
+# API Responsibilities
 
 Integrating the OmiseGO eWallet requires a new setup to be deployed. Feel free to [get in touch](mailto:thibault@omisego.co) for that step as we offer hosted solutions. Before starting any integration, it is important to understand which responsibilities OmiseGO is taking care of and which ones you will need to implement.
 
 ## Responsibilities
 
-### Features provided by the SDK
+### Features provided by the SDKs
```
Comment (T-Dnzt) [inline docs/guides/api_responsibilities.md]: Can we rename `SDK` to `eWallet` everywhere unless it's specifically about the Ruby/iOS/Android SDKs?

File: README.md [hunk 2]
```diff
@@ -1,440 +1,75 @@
...
+- [Issues](https://github.com/omisego/ewallet/issues): Browse or file a report for any bugs found
+- [Gitter](https://gitter.im/omisego/ewallet): Discuss features and suggestions in real-time
+- [StackOverflow](https://stackoverflow.com/questions/tagged/omisego): Search or create a new question with the tag `omisego`
+- Need enterprise support or hosting solutions? [Get in touch with us](mailto:thibault@omisego.co) for more details
 
-## Inserting some data
+## Community efforts
 
-Everything is in place and we can now run the seeds to populate the eWallet database with some initial data:
+We are thankful of our community for creating and maintaining these wonderful work that we otherwise could not have done ourselves. If you have ported any part of the OmiseGO eWallet to another platform, we will be very happy to list them here. [Submit us a pull request](pulls).
```
Comment (T-Dnzt) [inline README.md]: `thankful to our community`

File: docs/guides/guides.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Guides
+
+Take a deeper dive into the eWallet. Here are some resources to learn more about how the eWallet works.
+
+- [Usage](/docs/guides/usage.md): Start integrating your eWallet server to your application
+- [API Responsibilities](/docs/guides/api_responsibilities.md): Learn what responsibilities are taken care by the eWallet and what could be built on top of it.
+- [Entities](/docs/guides/entities.md): Learn the important terms used by the eWallet.
```
Comment (T-Dnzt) [inline docs/guides/guides.md]: Probably want to switch to `Terminology` based on the description.

Comment (unnawut) [reply docs/guides/guides.md]: Terminology probably should cover more than what we have in there. So I renamed terms -> entities instead.

File: README.md [hunk 4]
```diff
@@ -1,440 +1,75 @@
...
+- [Issues](https://github.com/omisego/ewallet/issues): Browse or file a report for any bugs found
+- [Gitter](https://gitter.im/omisego/ewallet): Discuss features and suggestions in real-time
+- [StackOverflow](https://stackoverflow.com/questions/tagged/omisego): Search or create a new question with the tag `omisego`
+- Need enterprise support or hosting solutions? [Get in touch with us](mailto:thibault@omisego.co) for more details
 
-## Inserting some data
+## Community efforts
 
-Everything is in place and we can now run the seeds to populate the eWallet database with some initial data:
+We are thankful to our community for creating and maintaining these wonderful work that we otherwise could not have done ourselves. If you have ported any part of the OmiseGO eWallet to another platform, we will be very happy to list them here. [Submit us a pull request](pulls).
```
Comment (mederic-p) [inline README.md]: I'm not sure that this link will work when merged on master.

**Pull request merged**